### ISSUE TITLE
`reactflow` in Remix app ---  SyntaxError: Named export 'Background' not found. The requested module '@reactflow/background' is a CommonJS module, which may not support all module.exports as named exports.

### ISSUE THREAD

**Issue by cliffordfajardo** [wbkd/react-flow#3238]
### Describe the Bug
When attempting to use reactflow in a remix app, I get the following error:

```
file:///Users/cfajardo/Downloads/react-flow-remix/build/index.js:212
import { ReactFlow } from "@reactflow/core";
         ^^^^^^^^^
SyntaxError: Named export 'ReactFlow' not found. The requested module '@reactflow/core' is a CommonJS module, which may not support all module.exports as named exports.
CommonJS modules can always be imported via the default export, for example using:

import pkg from '@reactflow/core';
const { ReactFlow } = pkg;

    at ModuleJob._instantiate (node:internal/modules/esm/module_job:128:21)
    at ModuleJob.run (node:internal/modules/esm/module_job:194:5)
    at async Promise.all (index 0)
    at ESMLoader.import (node:internal/modules/esm/loader:409:24)
    at file:///Users/cfajardo/Downloads/react-flow-remix/server.js:16:13
```

### Platform
- MacOS, Chrome
- Version: latest versions of react flow 11.x.x

### Your Example Website or App
- https://stackblitz.com/edit/node-fovhbz

### Steps to Reproduce the Bug or Issue
Create a fresh remix app (or [visit the stackblitz link](https://stackblitz.com/edit/node-fovhbz))

**1.Create a fresh remix app in your terminal:**
```
$ ~/Downloads npx create-remix@latest
? Where would you like to create your app? ./remix-reactflow
? What type of app do you want to create? Just the basics
? Where do you want to deploy? Choose Remix App Server if you're unsure; it's easy to change deployment targets.
Express Server
? TypeScript or JavaScript? TypeScript
? Do you want me to run `npm install`? Yes
⠇ Creating your app…
```

**2.Install `@reactflow` packages**
```
@reactflow/core @reactflow/background @reactflow/controls
```

3.Setup `root.tsx`  file with the content of the entire code snippet below ... this is to import the css for reactflow

<details>

```tsx
import { cssBundleHref } from "@remix-run/css-bundle";
import type { LinksFunction } from "@remix-run/node";
import {
  Links,
  LiveReload,
  Meta,
  Outlet,
  Scripts,
  ScrollRestoration,
} from "@remix-run/react";

import reactflowStyles from "@reactflow/core/dist/style.css";

export const links: LinksFunction = () => [
  ...(cssBundleHref ? [{ rel: "stylesheet", href: cssBundleHref }] : [
    { rel: "stylesheet", href: reactflowStyles },
  ]),
];

export default function App() {
  return (
    <html lang="en">
      <head>
        <meta charSet="utf-8" />
        <meta name="viewport" content="width=device-width,initial-scale=1" />
        <Meta />
        <Links />
      </head>
      <body>
        <Outlet />
        <ScrollRestoration />
        <Scripts />
        <LiveReload />
      </body>
    </html>
  );
}
```
</details>

**4. Copy this code into `app/routes/_index.tsx` to setup basic reactflow hello world app:**

<details>

```tsx
import type { V2_MetaFunction } from "@remix-run/node";
import {ReactFlow} from '@reactflow/core';


export const meta: V2_MetaFunction = () => {
  return [
    { title: "New Remix App" },
    { name: "description", content: "Welcome to Remix!" },
  ];
};

const initialNodes = [
  { id: '1', position: { x: 0, y: 0 }, data: { label: '1' } },
  { id: '2', position: { x: 0, y: 100 }, data: { label: '2' } },
];
const initialEdges = [{ id: 'e1-2', source: '1', target: '2' }];

export default function Index() {
  return (
    <div style={{ fontFamily: "system-ui, sans-serif", lineHeight: "1.8" }}>
      <h1>Welcome to Remix</h1>
      <div style={{ width: '100vw', height: '100vh' }}>
        <ReactFlow nodes={initialNodes} edges={initialEdges} />
    </div>
    </div>
  );
}

```
</details>

### Expected behavior

Code should work since:
`@reactflow` is using named exports according to these docs: https://reactflow.dev/docs/concepts/packages/#reactflowbackground

`@reactflow/{background, controls, core}` seem to be publishing ESM ([github repo](https://github.com/wbkd/react-flow/tree/main/packages))?
<img height="400" src="https://github.com/wbkd/react-flow/assets/6743796/027a4ccd-c801-45e7-ae87-72358df5ee04" />

## References / Related
- https://github.com/remix-run/remix/issues/6872
- https://stackoverflow.com/questions/65511127/when-can-a-commonjs-named-export-be-imported-by-an-es-module

**Comment by cliffordfajardo** [comment]
Hmm, this same error message happens with a fresh remix app when importing `lodash`
https://stackblitz.com/edit/node-s1pxmf?file=app%2Froutes%2F_index.tsx

<img width="1643" alt="image" src="https://github.com/remix-run/remix/assets/6743796/3002c552-4bc7-4a46-88d8-bb3da7c5ca62">

**Comment by bcakmakoglu** [comment]
According to [__publint__](https://publint.dev/@reactflow/core@11.7.4)
>pkg.module is used to output ESM, but pkg.exports is not defined. As NodeJS doesn't read pkg.module, the ESM output may be skipped. Consider adding pkg.exports to export the ESM output. pkg.module can usually be removed alongside too. (This will be a breaking change) ([More info](https://publint.dev/rules.html#has_module_but_no_exports))

**Comment by bcakmakoglu** [comment]
I'm assuming that's an issue for remix in this case? Though I'm not 100% sure about it

Though you can work around this problem by changing your `tsconfig.json` file and adding this

```json
    "paths": {
      "@reactflow/core": ["./node_modules/@reactflow/core/dist/esm"]
    },
```

[Here's a sandbox that works.](https://stackblitz.com/edit/node-aa97w8?file=app%2Froutes%2F_index.tsx,tsconfig.json)

**Comment by moklick** [comment]
Maybe this is related #2929 ?! I will have a look.

**Comment by moklick** [comment]
Could you try again? This should work with v11.8.0

**Comment by cliffordfajardo** [comment]
@moklick 

Looking at node_modules/@reactflow/core/dist/esm, there are both index.js and index.mjs in the esm folder. Switching the `module` field and the `exports.imports` field in the package.json file to point to `index.mjs` instead of `index.js` solved the issue for me. So this looks like misconfigured package.json in `@reactflow/*` and the main `reactflow` package.

**Comment by moklick** [comment]
I reverted back to .js, because .mjs breaks react 17 support (we are using "react-jsx" for the "jsx" compiler which doesn't seem to work with this setup)  :/

**Comment by moklick** [comment]
This should be fixed in 11.9.0
